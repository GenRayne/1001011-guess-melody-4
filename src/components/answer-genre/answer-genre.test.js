import React from 'react';
import renderer from 'react-test-renderer';
import AnswerGenre from './answer-genre.jsx';

const answer = {
  id: `1`,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  genre: `rock`,
};

const activePlayerId = `1`;
const userAnswer = true;
const noopHandler = () => {};

describe(`render AnswerGenre`, () => {
  it(`renders AnswerGenre`, () => {
    const tree = renderer
      .create(
          <AnswerGenre
            answer={answer}
            userAnswer={userAnswer}
            onAnswerChange={noopHandler}
            activePlayerId={activePlayerId}
            onPlayButtonClick={noopHandler}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
