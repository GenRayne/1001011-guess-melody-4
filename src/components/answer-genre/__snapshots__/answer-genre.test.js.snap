// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AnswerGenre renders AnswerGenre 1`] = `
<div
  className="track"
>
  <button
    aria-label="Pause"
    className="track__button track__button--pause"
    disabled={true}
    onClick={[Function]}
    title="Pause"
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
  <div
    className="game__answer"
  >
    <input
      checked={true}
      className="game__input visually-hidden"
      id="answer-1"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-1"
    />
    <label
      className="game__check"
      htmlFor="answer-1"
    >
      Отметить
    </label>
  </div>
</div>
`;
